docs/content/integrations/bigquery/using-bigquery-with-dagster.mdx

```diff
@@ -34,18 +47,195 @@ To complete this tutorial, you'll need:
+
+Once you have created an asset or source asset that represents a table in BigQuery, you will likely want to create additional assets that work with the data.
+
+```python file=/integrations/bigquery/tutorial/resource/downstream.py startafter=start_example endbefore=end_example
+from dagster import asset
+
+from .create_table import iris_data
+
+# this example uses the iris_dataset asset from Step 2
+
+
+@asset(deps=[iris_data])
+def iris_setosa(bigquery: BigQueryResource) -> None:
+    job_config = bq.QueryJobConfig(destination="iris.iris_setosa")
+    sql = "SELECT * FROM iris.iris_data WHERE species = 'Iris-setosa'"
+
+    with bigquery.get_client() as client:
+        job = client.query(sql, job_config=job_config)
+        job.result()
+```
+
+In this asset, you're creating second table that only contains the data for the _Iris Setosa_ species. This asset has a dependency on the `iris_data` asset. To define this dependency, you provide the `iris_data` asset as the `deps` parameter to the `iris_setosa` asset. You can then run the SQL query to create the table of _Iris Setosa_ data.
+
+### Completed code example
+
+When finished, your code should look like the following:
+
+```python file=/integrations/bigquery/tutorial/resource/full_example.py
+import pandas as pd
+from dagster_gcp import BigQueryResource
+from google.cloud import bigquery as bq
+
+from dagster import Definitions, SourceAsset, asset
+
+iris_harvest_data = SourceAsset(key="iris_harvest_data")
+
+
+@asset
+def iris_data(bigquery: BigQueryResource) -> None:
+    iris_df = pd.read_csv(
+        "https://docs.dagster.io/assets/iris.csv",
+        names=[
+            "sepal_length_cm",
+            "sepal_width_cm",
+            "petal_length_cm",
+            "petal_width_cm",
+            "species",
+        ],
+    )
+
+    with bigquery.get_client() as client:
+        job = client.load_table_from_dataframe(
+            dataframe=iris_df,
+            destination="iris.iris_data",
+        )
+        job.result()
+
+
+@asset(deps=[iris_data])
+def iris_setosa(bigquery: BigQueryResource) -> None:
+    job_config = bq.QueryJobConfig(destination="iris.iris_setosa")
+    sql = "SELECT * FROM iris.iris_data WHERE species = 'Iris-setosa'"
+
+    with bigquery.get_client() as client:
+        job = client.query(sql, job_config=job_config)
+        job.result()
+
+
+defs = Definitions(
+    assets=[iris_data, iris_setosa, iris_harvest_data],
+    resources={
+        "bigquery": BigQueryResource(
+            project="my-gcp-project",
+            location="us-east5",
+        )
+    },
+)
+```
+
+---
+
+## Part 2: Using the BigQuery I/O manager
+
+You may want to use an I/O manager to handle storing DataFrames as tables in BigQuery and loading BigQuery tables as DataFrames in downstream assets. Using an I/O manager is not required, and you can reference [When to use I/O managers](/concepts/io-management/io-managers#when-to-use-io-managers) to learn more.
```

Comment:
feeling we could probably clarify the why's for io managers more, such as:
- you want to pass data between assets
- you'd like to have dagster manage I/O of that data, aka how you store output and load it as input in the downstream (which would be helpful for re-execution case)